### ISSUE TITLE
Combination of email addresses

### ISSUE THREAD

**Issue by rfwolff** [devmount/third-stats#234]
Results for "Most received from" and "Most sent to" could be skewed if one person uses more than email address.
If so, it would be nice if several addresses could be linked to one person, e.g. via the address book.

**Comment by thetrial** [comment]
I just wanted to state the same. This would be a great achievement. Though I’m not quite sure yet how to combine, maybe even manually? But I have certain persons who used up to five or more accounts and would like to see them as one case.

**Comment by thetrial** [comment]
I’m not quite sure if this is practicable. I for myself use TB as a second programm on macOS. So TB does not use it’s own address book but system contacts. If all contacts with several mail adresses would contain all their past and alternative addresses those would be suggested when mailing. This could be disturbing. But maybe we could add an addressbook for collected and collated Mail addresses?

**Comment by devmount** [comment]
I see. In general I think this should be a feature of Thunderbird, to offer to mark email addresses of contacts as legacy, so that they won't be provided for new email recipients. In your special case: How does Thunderbird access your system contacts? Do you only use the "Collected Addresses"?

Providing functionality for manually connecting email addresses for contacts as add-on option is (at least for now) out of scope.

**Comment by thetrial** [comment]
I’m nbot sure how it accesses the contacts, but one can choose that TB uses the local address book. I guess this is an API, and it’s not so well done, because e.g. I use an alternative address book program. I don't use collected addresses (intentional). I use TB only as a second … mostly because of your plugin.

**Comment by devmount** [comment]
I feel honored that you keep using TB because of my add-on 😊 I'll issue a request for "address-bundles" at bugzilla.

**Comment by devmount** [comment]
Just for reference, I found a (20 year old!) feature request related to this at Bugzilla:
https://bugzilla.mozilla.org/show_bug.cgi?id=220370

**Comment by thetrial** [comment]
Well, kinda sad, isn’t it?

**Comment by devmount** [comment]
No, I don't think it is. As a FOSS maintainer myself, I know how it is if development fully depends on donations and to deal with people using your software for free and complaining about lack of features. No offense to you though, I don't know if and how much you supported/donated to TB 😇

**Comment by equaeghe** [comment]
> […] What I had in mind was to look up actual address book entries. So if you have contacts with multiple email addresses, you would still only have one entry in your address book for this person holding all the email addresses. […]

I think this option is the most practical. For those people that do not use an address book in Thunderbird, it should generally be possible to link their address book using Caldav (possibly as read-only) or via an VCS-export, to make it available in Thunderbird. (I use a CalDav address book in Thundrbird and mostly ignore the local Thunderbird address book.)

**Comment by devmount** [comment]
Thank you, Erik, for this addition. I think (or at least I hope), that with an address book webextension API it won't matter if you address book is local or from CalDAV / CardDAV. As long as Thunderbird can access it, it should be fine.

**Comment by devmount** [comment]
FYI: This would now be possible to implement. The API is there now: https://webextension-api.thunderbird.net/en/mv3/addressBooks.contacts.html
